File: my-folder/0069-sqrtx/solution.cpp
class Solution
{
public:
    int mySqrt(int x)
    {
        int a = 0;
        for (long long i = 0; i <= x; i++)
        {
            if ((i * i) == x)
            {
                return i;
            }

            else if ((i * i) > x)
            {

                return i - 1;
            }
        }
        return 0;
    }
};

File: my-folder/0069-sqrtx/solution.java
class Solution {
    public int mySqrt(int x) {
        if(x==0){
            return x;
        }
        int l = 1, r = x;
        while(l<=r){
            int mid = (r-l)/2+l;
            if(mid == x/mid){
                return mid;
            }
            else if(mid>x/mid){
                r = mid-1;
            }
            else{
                l = mid+1;
            }
        }
        return r;
    }
}
